### PR TITLE
Test setup

### PR DESCRIPTION
**Pull request by markuspf**
This adds the currently standard travis + codecov test setup to the package.

The authors should take care of the LICENSING issues, after merging travis and codecov have to be activated as well.

### CODE REVIEW

**Comment by olexandr-konovalov** [comment]
@markuspf what do you mean by "licensing issues" - there is a GPL file there. I was going to rename it to COPYING actually.

File: README.md [hunk 1]
```diff
@@ -16,3 +18,30 @@ The installation of the package is standard: for details, see GAP reference
 manual, chapter "GAP Packages" (and if you don't have write access to the
 Gap installation, type in GAP "?GAP root directory" to see another relevant
 part of the manual).
+
+## Documentation
+
+Full information and documentation can be found in the manual, available
+as PDF `doc/manual.pdf` or as HTML `htm/chapters.htm`, or on the package
```
Comment (olexandr-konovalov) [inline README.md]: From which package is this copied - I think I have already seen same corrections needed elsewhere. The path `htm/chapters.htm` is for gapmacro-based manuals.

File: README.md [hunk 2]
```diff
@@ -16,3 +18,30 @@ The installation of the package is standard: for details, see GAP reference
...
+
+License
+=======
+
+datastructures is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by the
+Free Software Foundation; either version 2 of the License, or (at your
+option) any later version.
+
+For details see the files COPYRIGHT.md and LICENSE.
```
Comment (olexandr-konovalov) [inline README.md:46]: I think this copyright info is provided elsewhere. Is there a recommendation for all packages to have now one more *.md file which COPYRIGHT.md ?

File: scripts/build_gap.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# build GAP in a subdirectory
+./autogen.sh
+./configure $GAP_CONFIGFLAGS
+make -j4 V=1
+
+# download packages; instruct wget to retry several times if the
+# connection is refused, to work around intermittent failures
+make bootstrap-pkg-full WGET="wget -N --no-check-certificate --tries=5 --waitretry=5 --retry-connrefused"
+
+git clone https://github.com/gap-packages/ferret pkg/ferret
```
Comment (olexandr-konovalov) [inline scripts/build_gap.sh]: There is no need in ferret for this package

File: scripts/build_pkg.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    ./autogen.sh
+    ./configure --with-gaproot=$GAPROOT
+    make -j4 V=1
+elif [[ -x configure ]]; then
+    ./configure $GAPROOT
+    make -j4
+fi
+
+# trick to allow the package directory to be used as a GAP root dir
+ln -s . pkg
```
Comment (olexandr-konovalov) [inline scripts/build_pkg.sh]: If this our latest standard setup (I mean, includes recent overhaul by @fingolfin) you need to look at this change that I would like to apply to it: https://github.com/gap-packages/example/pull/15/commits/786673dfaa153c5cf7dc2ac1d6e1bfbaec6e03a6

File: tst/testall.g [hunk 5]
```diff
@@ -1,31 +1,11 @@
...
+#
+# This file runs package tests. It is also referenced in the package
+# metadata in PackageInfo.g.
+#
+LoadPackage( "OpenMath" );
 
-# Arrange testfiles as required
-testfiles := [ "openmath02.tst", "openmath03.tst", "test_new" ];
+TestDirectory(DirectoriesPackageLibrary( "OpenMath", "tst" ),
+  rec(exitGAP := true));
```
Comment (olexandr-konovalov) [inline tst/testall.g]: Please keep using `rec(compareFunction := "uptowhitespace")` like it was before. There are derived tests (which have to be placed under version control, and then they will be tested by Travis and wrapped by releaseTools - it is an oversight now, as you see  "openmath02.tst", "openmath03.tst" are not in the repository). Manual examples may have pretty formatting and fail if the comparison is exact. 

You can generated them (as part of building the manual) and put under version control - then Travis CI will also test them. Although, I have a better version of examples extraction script in the Congruence package https://github.com/gap-packages/congruence - I've done this and some other changes around tests there just today. Sstarted to give a "spring clean" to my packages in alphabetic order :)

**Review by olexandr-konovalov — CHANGES_REQUESTED** [review]
Thanks @markuspf - I've left some notes in the comments.

**Comment by markuspf** [comment]
Tests pass again, I also added the manual examples to the testfiles.

File: tst/testall.g [hunk 6]
```diff
@@ -1,31 +1,12 @@
...
+# This file runs package tests. It is also referenced in the package
+# metadata in PackageInfo.g.
+#
+LoadPackage( "OpenMath" );
 
-# Arrange testfiles as required
-testfiles := [ "openmath02.tst", "openmath03.tst", "test_new" ];
+TestDirectory(DirectoriesPackageLibrary( "OpenMath", "tst" ),
+  rec( exitGAP := true
+     , testOptions := rec(compareFunction := "uptowhitespace")));
```
Comment (olexandr-konovalov) [inline tst/testall.g:9]: The comma on the new line... but if this is the only change in this PR, I can leave with this and fix this later.

**Pull request merged**